chimerapy/manager.py

```diff
@@ -446,6 +449,71 @@ def request_start(url):
 
         return all(success)
 
+    async def async_broadcast_request(
+        self,
+        htype: Literal["get", "post"],
+        route: str,
+        data: Any = {},
+        timeout: Union[int, float] = config.get("manager.timeout.info-request"),
+    ) -> bool:
+
+        # Create a new session for the moment
+        tasks: List[asyncio.Task] = []
+        sessions: List[aiohttp.ClientSession] = []
+        for worker_data in self.state.workers.values():
+            session = aiohttp.ClientSession()
+            url = f"http://{worker_data.ip}:{worker_data.port}" + route
+            logger.debug(f"{self}: Executing file transfer to {url}")
+            if htype == "get":
+                tasks.append(
+                    asyncio.create_task(session.get(url, data=json.dumps(data)))
+                )
+            elif htype == "post":
+                tasks.append(
+                    asyncio.create_task(session.post(url, data=json.dumps(data)))
+                )
+
+            # Storing sessions to later close
+            sessions.append(session)
+
+        outputs: List[aiohttp.ClientResponse] = await asyncio.gather(*tasks)
+
+        # Closing sessions
+        for session in sessions:
+            await session.close()
+
+        # Return if all outputs were successful
+        for output in outputs:
+            if output.status != 200:
+                return False
+
+        return True
+
+    async def async_collect(self) -> bool:
+
+        # Then, request to collect the archives
+        success = await self.async_broadcast_request(
+            htype="post",
+            route="/nodes/collect",
+            data={"path": str(self.logdir)},
+        )
+        self.state.collecting = False
+
+        if success:
+            self.server.move_transfer_files(self.logdir, True)
+            self.save_meta()
```

Comment:
Will a try-catch block be necessary here?

Reply:
Good points, a possible point of error